charts/cryostat/templates/deployment.yaml

```diff
@@ -34,69 +36,61 @@ spec:
-          {{- if not .Values.minimal }}
-          - name: GRAFANA_DATASOURCE_URL
-            value: http://127.0.0.1:8080
-          - name: GRAFANA_DASHBOARD_URL
-            value: "{{ if .Values.grafana.ingress.enabled }}http{{ if (or (.Values.grafana.ingress.tls) (.Values.grafana.sslProxied)) }}s{{ end }}://{{ with index .Values.grafana.ingress.hosts 0 }}{{ .host }}{{ end }}{{ end }}"
-          {{- end }}
-          - name: CRYOSTAT_DISABLE_SSL
-            value: "true"
-          - name: CRYOSTAT_DISABLE_JMX_AUTH
-            value: "true"
-          - name: CRYOSTAT_ALLOW_UNTRUSTED_SSL
-            value: "true"
-          - name: CRYOSTAT_ENABLE_JDP_BROADCAST
-            value: "false"
-          - name: CRYOSTAT_K8S_NAMESPACES
-            value: "{{ .Release.Namespace }}"
-          {{- if (or (and .Values.core.route.enabled .Values.core.route.tls.enabled) (and .Values.core.ingress.enabled .Values.core.ingress.tls) (.Values.core.sslProxied)) }}
-          - name: CRYOSTAT_SSL_PROXIED
+          - name: QUARKUS_HTTP_PROXY_PROXY_ADDRESS_FORWARDING
+            value: 'true'
+          - name: QUARKUS_HTTP_PROXY_ALLOW_X_FORWARDED
+            value: 'true'
+          - name: QUARKUS_HTTP_PROXY_ENABLE_FORWARDED_HOST
+            value: 'true'
+          - name: QUARKUS_HTTP_PROXY_ENABLE_FORWARDED_PREFIX
+            value: 'true'
+          - name: QUARKUS_HIBERNATE_ORM_DATABASE_GENERATION
+            value: drop-and-create
+          - name: QUARKUS_DATASOURCE_USERNAME
+            value: cryostat3
+          - name: QUARKUS_DATASOURCE_PASSWORD
+            valueFrom:
+              secretKeyRef:
+                name: {{ printf "%s-db-connection-key" .Release.Name }}
+                key: CONNECTION_KEY
+                optional: false
+          - name: QUARKUS_DATASOURCE_JDBC_URL
+            value: jdbc:postgresql://localhost:5432/cryostat3
+          - name: STORAGE_BUCKETS_ARCHIVES_NAME
+            value: archivedrecordings
+          - name: QUARKUS_S3_ENDPOINT_OVERRIDE
+            value: http://localhost:8333
+          - name: QUARKUS_S3_PATH_STYLE_ACCESS
             value: "true"
-          {{- end }}
-          {{- if ((.Values.pvc).enabled) }}
-          - name: CRYOSTAT_JDBC_URL
-            value: jdbc:h2:file:/opt/cryostat.d/conf.d/h2;INIT=create domain if not exists jsonb as varchar
-          - name: CRYOSTAT_HBM2DDL
-            value: update
-          - name: CRYOSTAT_JDBC_DRIVER
-            value: org.h2.Driver
-          - name: CRYOSTAT_HIBERNATE_DIALECT
-            value: org.hibernate.dialect.H2Dialect
-          - name: CRYOSTAT_JDBC_USERNAME
-            value: {{ include "cryostat.fullname" . }}
-          - name: CRYOSTAT_JDBC_PASSWORD
-            value: {{ include "cryostat.fullname" . }}
-          {{- end }}
-          - name: CRYOSTAT_JMX_CREDENTIALS_DB_PASSWORD
+          - name: QUARKUS_S3_AWS_REGION
+            value: us-east-1
```

Comment:
Is this just a dummy value for now? Does it do something with SeaweedFS?

Reply:
It doesn't actually do anything when using SeaweedFS, or probably when using any other "self-hosted" kind of S3 provider, but the SDK client wants this variable to be set to something - presumably because the actual AWS S3 requires it.

Reply:
https://docs.quarkiverse.io/quarkus-amazon-services/dev/amazon-s3.html#_configuring_s3_clients

> quarkus.s3.aws.region - It’s required by the client, but since you’re using a local S3 instance you can pick any valid AWS region.

Reply:
It would also be possible to hardcode a default value for this property in https://github.com/cryostatio/cryostat3/blob/main/src/main/resources/application.properties , so that end users can still override it if necessary but have a fallback if they don't, but I'm not sure if that's worthwhile. In the end if the user is going to choose some other S3 provider than the one we ship, they'll have to change all these related configuration knobs anyway.